vnext/src-win/Libraries/Text/Text.windows.js

```diff
@@ -294,6 +294,7 @@ const Text: React.AbstractComponent<
   } else {
     let styleProps: ViewStyleProp = (restProps.style: any);
     if (
+      global.RN$Bridgeless !== true && // [Windows] Fabric text handles borders, but on paper we need to wrap it in an extra view
```

Comment:
Is `global.RN$Bridgeless` the agreed upon JS check for new arch? Or just a convenient approximation?

Reply:
It's not quite an accurate check.  Core has specifically not exposed an agreed upon JS is fabric check, as core wants to avoid people doing JS checks for fabric.  We still have a test flag to switch bridge mode back to use the bridge in fabric which would cause this check to be wrong.  It is not a supported flag and requires magic strings to set - at this point I'm not sure if it even runs with that flag.  I added it in when we first moved to bridgeless to help verify if fallout from bridgeless was caused by bridgeless - but maybe we should remove it now that fabric bridge mode has been completely untested for quite a while.

Reply:
So based on core's opinion it sounds like there will never be a JS check for Fabric? Do we agree with that? If not, can we push for one? Maybe it's just Windows, and _maybe_ it's only an interim state while Fabric parity is not where Paper parity was and apps/modules need to navigate that... but a JS check would be quite useful.

Reply:
From Meta:  Happy to provide some context, as this question has been asked multiple times in the past.

In general, we'd like to avoid exposing a "isBridgeless" or "isFabricEnabled" public API, as we'd like to avoid split implementations. In general APIs that are working on Bridgeless are working also on Bridge (the inverse is not true though).

For example for Android, the correct way to dispatch events is to use the EventDispatcher. That API works on both Bridgeless and Bridge mode.
We'd like for folks to dont' branch their code for old/new arch and instead migrate to those newer APIs.

--- That's the philosophical vision ---

Realistically speaking then, more complex libraries, need to know if they're running on a Bridgeless/NewArch environment or not. I'd say that the RN$Bridgeless global is not going away and is ok to use (we use it also internally).
We'd like to remove it in the long run though